firebase-database/src/main/java/com/google/firebase/database/core/SyncTree.java

```diff
@@ -249,12 +249,12 @@ public boolean isEmpty() {
                 if (write.isOverwrite()) {
                   Node resolvedNode =
                       ServerValues.resolveDeferredValueSnapshot(
-                          write.getOverwrite(), existing, serverValues);
```

Comment:
I think you can delete "existing" above.

Reply:
Good catch. This also removed:

```
ArrayList<Long> excludeThis = new ArrayList<>();
excludeThis.add(write.getWriteId());
calcCompleteEventCache(write.getPath(), excludeThis);
```

That's a bit suspicious, but this never existing on iOS.